scipy/optimize/_milp.py

```diff
@@ -0,0 +1,280 @@
+        parameter of the `LinearConstraint` object is ignored.
+
+    Options
+    -------
+    presolve : bool (default: ``True``)
+        Presolve attempts to identify trivial infeasibilities,
+        identify trivial unboundedness, and simplify the problem before
+        sending it to the main solver. It is generally recommended
+        to keep the default setting ``True``; set to ``False`` if
+        presolve is to be disabled.
+    time_limit : float
+        The maximum time in seconds allotted to solve the problem;
+        default is the largest possible value for a ``double`` on the
+        platform.
+    dual_feasibility_tolerance : double (default: 1e-07)
+        Dual feasibility tolerance for
+        :ref:`'highs-ds' <optimize.linprog-highs-ds>`.
+        The minimum of this and ``primal_feasibility_tolerance``
+        is used for the feasibility tolerance of
+        :ref:`'highs-ipm' <optimize.linprog-highs-ipm>`.
+    primal_feasibility_tolerance : double (default: 1e-07)
+        Primal feasibility tolerance for
+        :ref:`'highs-ds' <optimize.linprog-highs-ds>`.
+        The minimum of this and ``dual_feasibility_tolerance``
+        is used for the feasibility tolerance of
+        :ref:`'highs-ipm' <optimize.linprog-highs-ipm>`.
+    ipm_optimality_tolerance : double (default: ``1e-08``)
+        Optimality tolerance for
+        :ref:`'highs-ipm' <optimize.linprog-highs-ipm>`.
+        Minimum allowable value is 1e-12.
+    simplex_dual_edge_weight_strategy : str (default: None)
+        Strategy for simplex dual edge weights. The default, ``None``,
+        automatically selects one of the following.
+
+        ``'dantzig'`` uses Dantzig's original strategy of choosing the most
+        negative reduced cost.
+
+        ``'devex'`` uses the strategy described in [15]_.
+
+        ``steepest`` uses the exact steepest edge strategy as described in
+        [16]_.
+
+        ``'steepest-devex'`` begins with the exact steepest edge strategy
+        until the computation is too costly or inexact and then switches to
+        the devex method.
+
+        Curently, ``None`` always selects ``'steepest-devex'``, but this
+        may change as new options become available.
+
+    Returns
+    -------
+    res : OptimizeResult
+        An instance of :class:`scipy.optimize.OptimizeResult`. The object
+        is guaranteed to have the following attributes.
+
+        status : int
+            An integer representing the exit status of the algorithm.
+        message : str
+            A string descriptor of the exit status of the algorithm.
+
+        Depending on the exit status, the following attributes may
+        also be available.
+
+        x : ndarray
+            The values of the decision variables that minimizes the
+            objective function while satisfying the constraints.
+        fun : float
+            The optimal value of the objective function ``c @ x``.
+        simplex_nit : int
```

Comment:
I think all `simplex_nit`/`ipm_nit`/`crossover_nit` will always be 0 (or -1 -- I think there might still be upstream thinking to do here) and aren't significant for MIP solutions.  We can instead report `mip_node_count`

Reply:
Is that from experimentation or did you find some documentation?

Reply:
I thought I had asked, but I can't seem to find the thread if I did.  In any case, yes -- empirically this what I find happens.  The counts are [set to `-1`](https://github.com/ERGO-Code/HiGHS/blob/06704da12bd057a5d59fcb3b7324b3994abcb5aa/src/lp_data/HighsInfo.cpp#L20) at the start of a problem and are never touched by the MIP solver, so stay at `-1`